File: python/7 kyu/336.py
# https://www.codewars.com/kata/6411b91a5e71b915d237332d/train/python

# Passed

def valid_parentheses(paren_str):
    balance = 0
    for char in paren_str:
        if char == "(":
            balance += 1
        elif char == ")":
            balance -=1
            
        if balance < 0:
            break
            
    result = balance == 0
    return result

output = valid_parentheses("(()())()")
print(output)

File: nodejs/8 kyu/336.js
// https://www.codewars.com/kata/563b74ddd19a3ad462000054/train/javascript

// Passed

function stringy(size) {
  let result = "";
  for (let idx = 0; idx < size; idx++) {
    result += Number(idx % 2 === 0);
  }

  return result;
}

const output = stringy(3);
console.log(output);